handler.go

```diff
@@ -173,50 +170,53 @@ func jsonRPCResponse(httpCode int, v interface{}) (*http.Response, error) {
 }
 
 func (t *myTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	lgr := t.lgr
+	ctx := req.Context()
 	if reqID := middleware.GetReqID(req.Context()); reqID != "" {
-		lgr = lgr.With(zap.String("requestID", reqID))
+		ctx = gotils.With(ctx, "requestID", reqID)
 	}
 
 	ip, methods, parsedRequests, err := parseRequests(req)
 	if err != nil {
-		lgr.Error("Failed to parse requests", zap.Error(err))
+		gotils.L(ctx).Error().Printf("Failed to parse requests: %v", err)
 		resp, err := jsonRPCResponse(http.StatusBadRequest, jsonRPCError(json.RawMessage("1"), jsonRPCInvalidParams, err.Error()))
 		if err != nil {
-			lgr.Error("Failed to construct invalid params response", zap.Error(err))
+			gotils.L(ctx).Error().Printf("Failed to construct invalid params response: %v", err)
 		}
 		return resp, nil
 	}
-	lgr = lgr.With(zap.String("remoteIp", ip), zap.Strings("methods", methods))
-	if blockResponse := t.block(req.Context(), lgr, parsedRequests); blockResponse != nil {
+
+	ctx = gotils.With(ctx, "remoteIp", ip)
+	ctx = gotils.With(ctx, "methods", methods)
+	if blockResponse := t.block(req.Context(), parsedRequests); blockResponse != nil {
```

Comment:
Existing `ctx` to keep fields?
```suggestion
	if blockResponse := t.block(ctx, parsedRequests); blockResponse != nil {
```